### ISSUE TITLE
Accessibility Audit of 10-10CG PDF Attachment

### ISSUE THREAD

**Issue by ala-yna** [department-of-veterans-affairs/va.gov-team#10559]
Given that we want the PDF to be 508 compliant
I want to audit the current implementation
So that we can make a plan to ensure accessibility.

**Comment by kevinmirc** [comment]
**Example 10-10CG Submissions as PDF**

Blank PDF: https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/pdf_fill/forms/pdfs/10-10CG.pdf

Filled in PDF (simple example): https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/fixtures/pdf_fill/10-10CG/simple.pdf

Filled in PDF (kitchen sink example): https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/fixtures/pdf_fill/10-10CG/kitchen_sink.pdf

Filled in PDF (overflow example): https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/fixtures/pdf_fill/10-10CG/overflow.pdf

**Comment by ala-yna** [comment]
Hello @jenstrickland! As a part of our desire to make sure our work is entirely accessible, we wanted to get your thoughts on our current PDF implementation, and your recommendations for making it the best it can be. I'm not sure if there's a formal Collab Cycle route for this, but here's the ticket, with the current implementation details in the above comments. Thanks for your help!

**Comment by jenstrickland** [comment]
Thanks, @ala-yna ! I'm going to mark this as a "Dev Intent" review point (exclusive to VSA teams, at this time). Starting on this now. 

Do you know where the base 10-10CG PDF came from, by chance?

**Comment by jenstrickland** [comment]
## 10-10CG Shell

### Document (4 issues)

#### Failed (We need to address)

- Primary Language
- Title

#### Needs manual check (We should check the document manually, but this isn't part of non-compliance errors.)

- Logical reading order
- Color contrast

### Page Content (3 issues)

#### Needs manual check

- Screen flicker
- Scripts
- Time responses

![Screen Shot 2020-07-06 at 10 52 00 AM](https://user-images.githubusercontent.com/57469/86607130-e1966f00-bf76-11ea-93be-86cb18618560.png)

![Screen Shot 2020-07-06 at 10 52 11 AM](https://user-images.githubusercontent.com/57469/86607145-e5c28c80-bf76-11ea-8c09-64005141bc71.png)

![Screen Shot 2020-07-06 at 10 52 22 AM](https://user-images.githubusercontent.com/57469/86607154-e9eeaa00-bf76-11ea-868f-b0f23935270b.png)

**Comment by jenstrickland** [comment]
## Simple

The same issues as the shell.

![Screen Shot 2020-07-06 at 1 11 22 PM](https://user-images.githubusercontent.com/57469/86620486-63dc5e80-bf8a-11ea-8e10-4889db7b8f76.png)

![Screen Shot 2020-07-06 at 1 11 33 PM](https://user-images.githubusercontent.com/57469/86620498-6939a900-bf8a-11ea-9795-1656c53c32dd.png)

![Screen Shot 2020-07-06 at 1 11 45 PM](https://user-images.githubusercontent.com/57469/86620506-6ccd3000-bf8a-11ea-9bae-67ec48eaafa0.png)

**Comment by jenstrickland** [comment]
## Kitchen Sink

The same issues as the shell.

![Screen Shot 2020-07-06 at 1 14 19 PM](https://user-images.githubusercontent.com/57469/86620667-b87fd980-bf8a-11ea-9818-a77bbc7da53e.png)

![Screen Shot 2020-07-06 at 1 14 28 PM](https://user-images.githubusercontent.com/57469/86620680-be75ba80-bf8a-11ea-83f6-2b88a7ef4cd5.png)

![Screen Shot 2020-07-06 at 1 14 37 PM](https://user-images.githubusercontent.com/57469/86620690-c170ab00-bf8a-11ea-8bbc-43868084743b.png)

**Comment by jenstrickland** [comment]
## Overflow

The same issues as the shell.

![Screen Shot 2020-07-06 at 1 16 15 PM](https://user-images.githubusercontent.com/57469/86620844-13b1cc00-bf8b-11ea-922c-783f677e06f8.png)

![Screen Shot 2020-07-06 at 1 17 12 PM](https://user-images.githubusercontent.com/57469/86620851-16acbc80-bf8b-11ea-8415-a16a6b395dec.png)

![Screen Shot 2020-07-06 at 1 17 23 PM](https://user-images.githubusercontent.com/57469/86620855-19a7ad00-bf8b-11ea-9b83-22f427076e56.png)

**Comment by jenstrickland** [comment]
## Overflow + extras

This document is very different from the initial form.

### Document (5 issues)

#### Failed

- Tagged PDF
- Primary Language
- Title

#### Needs manual check

- Logical reading order
- Color contrast

### Page Content (2 issues)

#### Failed

- Tagged content
- Tab order


### Alternate Text (5 issues)

#### Failed

- Figures alternate text
- Nested alternate text
- Associated with content
- Hides annotation
- Other elements alternate text


### Tables (4 issues)

#### Failed

- Rows
- TH and TD
- Headers
- Regularity


### Lists (2 issues)

#### Failed

- List items
- Lbl and LBody


### Headings (1 issue)

#### Failed

- Appropriate nesting


![Screen Shot 2020-07-06 at 1 18 35 PM](https://user-images.githubusercontent.com/57469/86621167-b8340e00-bf8b-11ea-9fd7-699dff09c641.png)

![Screen Shot 2020-07-06 at 1 18 51 PM](https://user-images.githubusercontent.com/57469/86621177-bc602b80-bf8b-11ea-8476-6189618fd680.png)

![Screen Shot 2020-07-06 at 1 19 04 PM](https://user-images.githubusercontent.com/57469/86621184-bf5b1c00-bf8b-11ea-9908-6f2e0b071db7.png)

![Screen Shot 2020-07-06 at 1 19 15 PM](https://user-images.githubusercontent.com/57469/86621188-c1bd7600-bf8b-11ea-98ed-c6a903464f8f.png)

![Screen Shot 2020-07-06 at 1 19 25 PM](https://user-images.githubusercontent.com/57469/86621200-c550fd00-bf8b-11ea-9aae-938839030e6b.png)

![Screen Shot 2020-07-06 at 1 19 34 PM](https://user-images.githubusercontent.com/57469/86621206-c84bed80-bf8b-11ea-9bb6-b43d615780b6.png)

![Screen Shot 2020-07-06 at 1 19 43 PM](https://user-images.githubusercontent.com/57469/86621214-cb46de00-bf8b-11ea-968f-0316bb33919f.png)

![Screen Shot 2020-07-06 at 1 20 22 PM](https://user-images.githubusercontent.com/57469/86621216-ce41ce80-bf8b-11ea-8f85-42fbebcf3c8f.png)

![Screen Shot 2020-07-06 at 1 20 32 PM](https://user-images.githubusercontent.com/57469/86621221-d13cbf00-bf8b-11ea-9e08-d23e849fa0a7.png)

**Comment by jenstrickland** [comment]
@kevinmirc Thanks for the PDF examples. Do you have any more information on where the base file for the Overflow_Extras came from? 

I'm wondering if that is a file used by the PDF generation utility for when the Additional Information scenario comes up.

cc @ala-yna

**Comment by jenstrickland** [comment]
## Recommendations to ensure 508 compliance

1. Ensure the 10-10CG shell PDF is compliant to begin with. The assumption here is that when the data populates the shell it would be compliant, as shown in these examples.
1. Locate the origin of the Additional Information shell, and ensure the shell is compliant. The same hypothesis applies.

**Comment by kevinmirc** [comment]
Hi @jenstrickland, the Additional Information page is generated based on our custom implementation written in vets-api. This can be customized in `lib/pdf_fill/extras_generator.rb`.

For context: Developers who add a PDF to the PDF Fill must specify a maximum length and label for each PDF field. If the value of a field exceeds the maximum length defined, it will add the label and value to the Additional Information sec tion.

So we (VSA) have full control over how we format and style the Additional Infomation page.

**Comment by jenstrickland** [comment]
Thanks, @kevinmirc!

When the Additional Information page is used, does it follow the same process as the other PDFs? Based on my audit, it seems there is a base "shell" PDF, information is appended to that, and a new file saved. The same errors in all the original shell files were exactly the same errors in the new file. I just don't have the base for the Additional Information page. I'm not concerned with the format/style, but the 508 compliance of the file. My hypothesis is there is a base file for Additional Information that apps add information to. That base shell should be compliant — and is likely the responsibility of where the file started. Perhaps we should hop on a call to discuss. Please find some time on my calendar next week so we can clarify.

**Comment by ala-yna** [comment]
The PDF that we're using to generate the 10-10CG can be found on this page:https://www.caregiver.va.gov/support/support_benefits.asp

The specific link to the form is here: https://www.va.gov/vaforms/medical/pdf/10-10CG.pdf#

We were sent a copy by Pamela Wright, LCSW-R, VACO, the National Program Manager for the VA Caregiver Support Line on April 2, 2020.

**Comment by jenstrickland** [comment]
For the Additional Information page, that is generated by an `extras_generator.rb` file ( https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/pdf_fill/extras_generator.rb ), which creates extra pages when submitted PDFs run long. This PDF is currently not 508-compliant because the script does not generated a tagged PDF. VSP (@1Copenut ) and @kevinmirc are checking to see if VSP is okay with VSA looking into a fix. We need to know if this is work VSP wants to handle or not - see [Slack discussion](https://dsva.slack.com/archives/CBU0KDSB1/p1596469767430400).

**Comment by jenstrickland** [comment]
Just sent the base shell PDF to the VA 508 Office for them to contact the PDF owner to remediate the 10-10CG base PDF.
